src/pages/QuizPage.tsx

```diff
@@ -1,18 +1,32 @@
 import './QuizPage.css';
 import { ButtonComponent } from '../components/ButtonComponent';
 import { InputComponent } from '../components/InputComponent';
-import { useState } from 'react';
+import { useEffect, useState } from 'react';
 import { Link, useNavigate } from 'react-router-dom';
 import ModalComponent from '../components/ModalComponent';
+import { useDispatch, useSelector } from 'react-redux';
+import { RootState, useFetchQuestionsQuery } from '../store/reducers';
+import { increaseQuestionIndex, setQuestions } from '../store/questionsSlice';
 
 const QuizPage = () => {
   const [modalIsOpen, setModalIsOpen] = useState(false);
   const navigate = useNavigate();
-  const timer = 0;
+  const { countQuestions, category, difficulty, type } = useSelector((state: RootState) => state.config);
+  const { questions, questionsIndex } = useSelector((state: RootState) => state.questions);
+  const { data, isFetching } = useFetchQuestionsQuery({ countQuestions, category, difficulty, type });
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    if (data?.results) {
+      dispatch(setQuestions(data.results));
+    }
+  }, [data, dispatch]);
 
   function nextAction() {
-    if (!timer) {
+    if (questionsIndex + 1 === data?.results.length) {
       navigate('/result');
+    } else {
+      dispatch(increaseQuestionIndex());
     }
   }
```

Comment:
most of this code might be extracted into a new custom hook which returns some values you use in this component
it makes this component cleaner 👍